qiskit/aqua/operators/gradients/derivative_base.py

```diff
@@ -0,0 +1,249 @@
+        r"""
+        Args:
+            operator: The operator we are taking the gradient, Hessian or QFI of
+            params: The parameters we are taking the gradient, Hessian or QFI with respect to.
+
+        Returns:
+            An operator whose evaluation yields the gradient, Hessian or QFI.
+
+        Raises:
+            ValueError: If ``params`` contains a parameter not present in ``operator``.
+        """
+        raise NotImplementedError
+
+    def gradient_wrapper(self,
+                         operator: OperatorBase,
+                         bind_params: Union[ParameterExpression, ParameterVector,
+                                            List[ParameterExpression]],
+                         grad_params: Optional[Union[ParameterExpression, ParameterVector,
+                                                     List[ParameterExpression],
+                                                     Tuple[ParameterExpression,
+                                                           ParameterExpression],
+                                                     List[Tuple[ParameterExpression,
+                                                                ParameterExpression]]]] = None,
+                         backend: Optional[Union[BaseBackend, QuantumInstance]] = None) \
+            -> Callable[[Iterable], np.ndarray]:
+        """Get a callable function which provides the respective gradient, Hessian or QFI for given
+        parameter values. This callable can be used as gradient function for optimizers.
+
+        Args:
+            operator: The operator for which we want to get the gradient, Hessian or QFI.
+            bind_params: The operator parameters to which the parameter values are assigned.
+            grad_params: The parameters with respect to which we are taking the gradient, Hessian
+                        or QFI. If grad_params = None, then grad_params = bind_params
+            backend: The quantum backend or QuantumInstance to use to evaluate the gradient,
+                Hessian or QFI.
+
+        Returns:
+            callable(param_values): Function to compute a gradient, Hessian or QFI. The function
+            takes an iterable as argument which holds the parameter values.
+
+        Raises:
+            ImportError: if 'retworkx>=0.5.0' is not installed.
+        """
+        if not grad_params:
+            grad_params = bind_params
+        if backend and (
+                (isinstance(backend, QuantumInstance) and (not backend.is_statevector)) or
+                (not isinstance(backend, QuantumInstance) and
+                 not backend.name().startswith('statevector'))
+        ):
+            try:
+                import retworkx
+            except ImportError as ex:
+                raise ImportError('Please install retworx>=0.5.0') from ex
+
+            if retworkx.__version__ < '0.5.0':
+                raise ImportError('Please update retworx to at least version 0.5.0')
+
```

Comment:
Aqua requires retworkx as a dependent listed in setup.py where it lists 0.5.0. This check here is unneccesary and should ideally be removed as retworkx will be installed as dependent when Aqua is installed.